source/create.ts

```diff
@@ -99,7 +100,29 @@ const aliases: readonly HTTPAlias[] = [
 	'delete'
 ];
 
-export const defaultHandler: HandlerFunction = (options, next) => next(options);
+export const defaultHandler: HandlerFunction = (options, next) => {
+	const result = next(options);
+
+	if (is.nodeStream(options.body)) {
+		const errorHandler = (error: Error): void => {
+			if (options.isStream) {
+				(result as ProxyStream).destroy(error);
+				return;
+			}
+
+			(result as CancelableRequest<unknown>).cancel(error.message);
+		};
+
+		options.body.once('error', errorHandler);
+
+		// @ts-ignore Each member of the union type '...' has signatures, but none of those signatures are compatible with each other
```

Comment:
I think it just says that result can be either or a stream or a promise and it doesn't know what types to choose